cmd/gitops-server/cmd/cmd.go

```diff
@@ -211,7 +214,15 @@ func runCmd(cmd *cobra.Command, args []string) error {
 		),
 	)
 
-	coreConfig := core.NewCoreConfig(log, rest, cacheContainer, clusterName)
+	fetcher, err := fetcher.NewSingleClusterFetcher(rest)
+	if err != nil {
+		return err
+	}
+
+	clusterClientsFactory := clustersmngr.NewClientFactory(rawClient, fetcher, nsaccess.NewChecker(nsaccess.DefautltWegoAppRules), log)
+	clusterClientsFactory.Start(context.TODO())
+
+	coreConfig := core.NewCoreConfig(log, rest, cacheContainer, clusterName, clusterClientsFactory)
```

Comment:
Do we still need the `cacheContainer`?

Reply:
we don't need it anymore, I was planning to do the clean-up of unused code in another PR to minimize the size of this one which is already big enough.